plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java

```diff
@@ -780,18 +785,125 @@ public void notifyVersionUpgrade(@NotNull Version currentVersion, @NotNull Versi
         dialog.open();
     }
 
-    public void setResetUIOnRestart(boolean resetUIOnRestart) {
-        this.resetUIOnRestart = resetUIOnRestart;
+    public void setResetUserPreferencesOnRestart(boolean resetUserPreferencesOnRestart) {
+        this.resetUserPreferencesOnRestart = resetUserPreferencesOnRestart;
     }
 
-    public void setResetWorkspaceOnRestart(boolean resetWorkspaceOnRestart) {
-        this.resetWorkspaceOnRestart = resetWorkspaceOnRestart;
+    public void setResetWorkspaceConfigurationOnRestart(boolean resetWorkspaceConfigurationOnRestart) {
+        this.resetWorkspaceConfigurationOnRestart = resetWorkspaceConfigurationOnRestart;
     }
 
     protected void setIgnoreRecentWorkspaces(boolean ignoreRecentWorkspaces) {
         this.ignoreRecentWorkspaces = ignoreRecentWorkspaces;
     }
 
+    private void loadStartupActions(@NotNull Location instanceLoc) {
+        final Path path = GeneralUtils.getMetadataFolder().resolve(STARTUP_ACTIONS_FILE);
+
+        if (Files.notExists(path)) {
+            return;
+        }
+
+        try (Reader reader = Files.newBufferedReader(path)) {
+            final Properties properties = new Properties();
+            properties.load(reader);
+
+            if (!properties.isEmpty()) {
+                processStartupActions(instanceLoc, properties.stringPropertyNames());
+            }
+        } catch (Exception e) {
+            log.error("Unable to read startup actions", e);
+        } finally {
+            try {
+                Files.delete(path);
+            } catch (IOException e) {
+                log.error("Unable to delete startup actions file: " + e.getMessage());
+            }
+        }
+    }
+
+    private void saveStartupActions() {
+        final Properties props = new Properties();
+
+        if (resetWorkspaceConfigurationOnRestart) {
+            props.setProperty(RESET_WORKSPACE_CONFIGURATION, Boolean.TRUE.toString());
+        }
+
+        if (resetUserPreferencesOnRestart) {
+            props.setProperty(RESET_USER_PREFERENCES, Boolean.TRUE.toString());
+        }
+
+        if (!props.isEmpty()) {
+            try (Writer writer = Files.newBufferedWriter(GeneralUtils.getMetadataFolder().resolve(STARTUP_ACTIONS_FILE))) {
+                props.store(writer, "DBeaver startup actions");
+            } catch (Exception e) {
+                log.error("Unable to save startup actions", e);
+            }
+        }
+    }
+
+    private void processStartupActions(@NotNull Location instanceLoc, @NotNull Set<String> actions) throws Exception {
+        final boolean resetUserPreferences = actions.contains(RESET_USER_PREFERENCES);
+        final boolean resetWorkspaceConfiguration = actions.contains(RESET_WORKSPACE_CONFIGURATION);
+
+        if (!resetUserPreferences && !resetWorkspaceConfiguration || !instanceLoc.isSet()) {
+            return;
+        }
+
+        final Path path = Path.of(instanceLoc.getDataArea("").toURI());
```

Comment:
You probably don't want to use toURI(), but I'm not 100% sure. @Destrolaric is an expert in this now.